pkg/api/customization/cluster/formatter.go

```diff
@@ -47,7 +47,23 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 			resource.AddAction(request, v3.ClusterActionBackupEtcd)
 			resource.AddAction(request, v3.ClusterActionRestoreFromEtcdBackup)
 		}
-		resource.AddAction(request, v3.ClusterActionRunSecurityScan)
+		canRunScan := func() bool {
+			revision := map[string]interface{}{
+				"id": request.ID,
+			}
+
+			return request.AccessControl.CanDo(
+				v3.ClusterScanGroupVersionKind.Group,
+				v3.ClusterScanResource.Name,
+				"create",
+				request,
+				revision,
+				request.Schema,
+			) == nil
```

Comment:
I am not sure if we can check "create" ability using CanDo. That method sound like more for checking if an action/verb can be applied to an existing object as pointed by the obj map[string]interface{} parameter.

Reply:
Maybe we need to do something like https://github.com/rancher/rancher/blob/master/pkg/api/customization/cluster/actionhandler.go#L48-L53

Reply:
As discussed offline, runSecurityScan behind the scenes creates a clusterscan object. Hence checking for "create" action.